viewslides.py

```diff
@@ -708,28 +650,16 @@ def add_image(self):
                 str(arcname) +
                 ' already exists in slideshow!')
             return
-        # Assign a file path to create if one doesn't exist yet
-        if self.tempfile is None:
-            self.tempfile = os.path.join(self.get_activity_root(), 'instance',
-                                         'tmp%i' % time.time())
-        try:
-            if os.path.exists(self.tempfile):
-                zf = zipfile.ZipFile(self.tempfile, 'a')
-            else:
-                zf = zipfile.ZipFile(self.tempfile, 'w')
-            zf.write(selected_file.encode("utf-8"), arcname.encode("utf-8"))
-            zf.close()
-            iter = self.ls_left.append()
-            self.ls_left.set(
-                iter,
-                COLUMN_IMAGE,
-                arcname,
-                COLUMN_OLD_NAME,
-                arcname)
-            self._slides_toolbar._add_image.props.sensitive = False
-        except BadZipfile as err:
-            print 'Error opening the zip file: %s' % (err)
-            self._alert('Error', 'Error opening the zip file')
+
+        iter = self.ls_left.append()
+        self.ls_left.set(
+            iter,
+            COLUMN_IMAGE,
+            arcname,
+            COLUMN_OLD_NAME,
+            arcname)
```

Comment:
It would be nice to use some other variable name other than `iter` which would shadow the built-in function `iter`

```suggestion
        selection_iter = self.ls_left.append()
        self.ls_left.set(
            selection_iter,
            COLUMN_IMAGE,
            arcname,
            COLUMN_OLD_NAME,
            arcname)
```

Reply:
Thanks for pointing that out, fixed in 6077947.